WIP: Feature/gantt refactor and tests

kingy04 opened the pull request:
This PR includes a cleanup on the gantt component, which had quite a few local data props and some component props which were not used among other bits of code etc which could be removed (as far as i could see)

It also brings in jest and some test coverage focusing at first on the latest logic recently added for building out the gantt data for rendering events by groups etc. It breaks some of the bulkier methods down to improve testability

Hopefully this will provide us with a good base to add to when adding new features and for adding more tests for some of the older logic when refactoring etc

It also clears out some old testing packages from dev dependencies

If you feel i have removed anything that should have stayed, or have missed the mark with some of the tests please let me know your thoughts, this is ultimately why this is a WIP PR.

PS. this PR includes the fixes #18, we should merge that in before merging this in.

BrockReece commented:
I have just run these tests locally and some of the snapshot tests have failed, it looks like the start and end dates have moved forward by a day.

kingy04 commented:
Thanks @BrockReece, I forget about the done argument TBH, i have updated the aysnc tests and tested each one making sure it fails first.. thats something i have been trying to cement into my workflow. I have added some static offsets / dates in the test data rather than the dynamically generated moment values which was the problem.. those should stay put now, i have tested that by changing my machines date a few times and running the tests. Cheers
